src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+                Camera* cam = Gameplay_GetCamera(globalCtx, 0);
+
+                cam->eye = this->subCamEye;
+                cam->eyeNext = this->subCamEye;
+                cam->at = this->subCamAt;
+                func_800C08AC(globalCtx, this->subCamId, 0);
+                this->subCamId = 0;
+                this->csState2 = this->subCamId;
+                func_80064534(globalCtx, &globalCtx->csCtx);
+                func_8002DF54(globalCtx, &this->actor, 7);
+                this->work[TW_PLLR_IDX] = 0;
+                this->targetPos = sTwinrovaPillarPos[0];
+                BossTw_TwinrovaSetupFly(this, globalCtx);
+            }
+            break;
+    }
+}
+
+void BossTw_SetupDeathCS(BossTw* this, GlobalContext* globalCtx) {
+    this->actionFunc = BossTw_DeathCS;
+    Animation_MorphToLoop(&this->skelAnime, &D_060004A4, -3.0f);
+    this->unk_5F8 = 0;
+    this->work[CS_TIMER_2] = Rand_ZeroFloat(20.0f);
+}
+
+void BossTw_DeathCS(BossTw* this, GlobalContext* globalCtx) {
+    if (this->timers[0] == 0) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->work[YAW_TGT], 5, this->rotateSpeed);
+    Math_ApproachF(&this->rotateSpeed, 20480.0f, 1.0f, 1000.0f);
+
+    if (sTwinrovaPtr->work[CS_TIMER_2] > 140) {
+        Math_ApproachF(&this->fogR, 100.0f, 1.0f, 15.0f);
+        Math_ApproachF(&this->fogG, 255.0f, 1.0f, 15.0f);
+        Math_ApproachF(&this->fogB, 255.0f, 1.0f, 15.0f);
+        Math_ApproachF(&this->fogNear, 850.0f, 1.0f, 15.0f);
+        Math_ApproachF(&this->fogFar, 1099.0f, 1.0f, 15.0f);
+    }
+}
+
+void BossTw_SetupCSWait(BossTw* this, GlobalContext* globalCtx) {
+    this->actionFunc = BossTw_CSWait;
+    this->visible = false;
+    this->actor.world.pos.y = -2000.0f;
+    this->actor.flags &= ~1;
+}
+
+/**
+ * Do nothing while waiting for the inital cutscene to start
+ */
+void BossTw_CSWait(BossTw* this, GlobalContext* globalCtx) {
+}
+
+void BossTw_TwinrovaSetupIntroCS(BossTw* this, GlobalContext* globalCtx) {
+    this->actionFunc = BossTw_TwinrovaIntroCS;
+    this->visible = false;
+    this->actor.world.pos.y = -2000.0f;
+    this->actor.flags &= ~1;
+}
+
+void BossTw_TwinrovaIntroCS(BossTw* this, GlobalContext* globalCtx) {
+    u8 updateCam = 0;
+    s16 i;
+    Vec3f sp90;
+    Vec3f sp84;
+    Player* player = PLAYER;
+
+    if (this->csSfxTimer > 220 && this->csSfxTimer < 630) {
+        func_80078884(NA_SE_EN_TWINROBA_UNARI - SFX_FLAG);
+    }
+
+    if (this->csSfxTimer == 180) {
+        func_80078914(&D_8094A7D0, NA_SE_EN_TWINROBA_LAUGH);
+        func_80078914(&D_8094A7D0, NA_SE_EN_TWINROBA_LAUGH2);
+        Audio_QueueSeqCmd(0x61);
+    }
+
+    this->csSfxTimer++;
+
+    switch (this->csState2) {
+        case 0:
+            this->csSfxTimer = 0;
+
+            if (SQ(player->actor.world.pos.x) + SQ(player->actor.world.pos.z) < 22500.0f) {
+                player->actor.world.pos.x = player->actor.world.pos.z = .0f;
```

Comment:
```suggestion
                player->actor.world.pos.x = player->actor.world.pos.z = 0.0f;
```
inb4 `.0f` is required for matching

Reply:
youd be correct. the one in the comment below was fine to change tho lol